Remotion/ObjectBinding/Validation/BusinessObjectValidationResult.cs

```diff
@@ -50,23 +50,62 @@ private static BusinessObjectValidationFailure CreateBusinessObjectValidationFai
             when validationFailure.ValidatedObject is IBusinessObject validatedBusinessObject
                  && GetBusinessObjectProperty(validatedBusinessObject, propertyValidationFailure.ValidatedProperty) is IBusinessObjectProperty validatedBusinessObjectProperty:
         {
-          return BusinessObjectValidationFailure.CreateForBusinessObjectProperty(
+          yield return BusinessObjectValidationFailure.CreateForBusinessObjectProperty(
               propertyValidationFailure.LocalizedValidationMessage,
               validatedBusinessObject,
               validatedBusinessObjectProperty);
+
+          break;
+        }
+
+        case ObjectValidationFailure objectValidationFailure
+            when objectValidationFailure.AffectedProperties.Count > 0:
+        {
+          var anyFailureReported = false;
+          foreach (var affectedProperty in objectValidationFailure.AffectedProperties)
+          {
+            if (affectedProperty.RelatedObject is not IBusinessObject validatedBusinessObject)
+              continue;
+
+            if (affectedProperty.Property != null
+                && GetBusinessObjectProperty(validatedBusinessObject, affectedProperty.Property) is IBusinessObjectProperty validatedBusinessObjectProperty)
+            {
+              yield return BusinessObjectValidationFailure.CreateForBusinessObjectProperty(
+                  objectValidationFailure.LocalizedValidationMessage,
+                  validatedBusinessObject,
+                  validatedBusinessObjectProperty);
+            }
+            else
+            {
+              yield return BusinessObjectValidationFailure.CreateForBusinessObject(
+                  validatedBusinessObject,
+                  objectValidationFailure.LocalizedValidationMessage);
+            }
+
+            anyFailureReported = true;
+          }
+
+          if (!anyFailureReported)
```

Comment:
Not sure about the best design for the empty option when doing this with a sequence, perhaps the cleanest option is to go with ApplySideEffect(()=> hasFailuresReported = true)

=> `.properties.Where (item=>item.RelatedObject is IBusienssObject).ApplySideEffect(_=>any=true).Select(item=>GetResult(item)`